### DISCUSSION TITLE
Problem with @action class decorator- missing context parameter

### DISCUSSION THREAD

**Discussion by TomMarks** [elgatosf/elgato#132] [Help] [answered]
I've scaffolded out a basic project with "streamdeck create" and gotten the code to start compiling, but I get an error message when compiling.  Below is an excerpt from the top of increment-counter.ts file created by streamdeck create.
```
import { action, KeyDownEvent, SingletonAction, WillAppearEvent } from "@elgato/streamdeck";

/**
 * An example action class that displays a count that increments by one each time the button is pressed.
 */
@action({ UUID: "com.test.test.increment" })
export class IncrementCounter extends SingletonAction<CounterSettings> {

```

The error returned by the typescript compiler is below.  This must be a configuration problem, but can't find any guidance anywhere.  Any ideas would be welcomed!

```
src/plugin.ts → com.test.test.sdPlugin/bin/plugin.js...
(!) [plugin typescript] src/actions/increment-counter.ts (6:1): @rollup/plugin-typescript TS1238: Unable to resolve signature of class decorator when called as an expression.
  The runtime will invoke the decorator with 1 arguments, but the decorator expects 2.
E:/OneDrive/Dev/test/src/actions/increment-counter.ts:6:1
 expression.
  The runtime will invoke the decorator with 1 arguments, but the decorator expects 2.

6 @action({ UUID: "com.test.test.increment" })
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

  node_modules/@elgato/streamdeck/dist/index.d.ts:3024:116
    3024 declare function action(definition: ActionDefinition): <T extends Constructor<SingletonAction<object>>>(target: T, context: ClassDecoratorContext) => {
                                                                                                                            
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    An argument for 'context' was not provided.

```
[tsconfig.json](https://github.com/user-attachments/files/17067561/tsconfig.json)
[package.json](https://github.com/user-attachments/files/17067565/package.json)

**Comment by GeekyEggo** [comment]
Hi Tom, apologies for the delay — so we can best help with your issue, it would be good to know a bit more about which versions you're running locally. Specifically, it would be good to know:

- The version of Node.js
  - Found by running `node -v` in terminal.
- The versions of `@elgato/streamdeck` and `typescript` the plugin project is using
  - These can be found by running `npm list --depth=0` in the cwd of your plugin project.
- The version of Stream Deck app.
  - Accessible within the preferences pane.

**Comment by TomMarks** [comment]
No worries- Thanks for getting back to me.  I'm excited to get this working!

Node version: v22.9.0
@elgato/cli: 0.3.2
@elgato/streamdeck: 0.3.0
Streamdeck App: 6.7.3 (21005)

**Reply by GeekyEggo** [reply]
Those versions look okay; we would recommend building with Node.js v20 as this is distributed with Stream Deck app, however testing locally with v22.9.0, that doesn't look to be the issue.

Could I ask what version of TypeScript you're using?

**Reply by GeekyEggo** [reply, marked as answer]
ah, I think I've found the issue — could I ask you to remove `"experimentalDecorators": true` from your tsconfig and try again 🤞

**Reply by TomMarks** [reply]
Hey GE- that did it.  Thanks!
